pallets/phala-world/src/lib.rs

```diff
@@ -0,0 +1,814 @@
+		/// authenticated claimer
+		///
+		/// Parameters:
+		/// - origin: The origin of the extrinsic.
+		/// - serial_id: The serial id of the spirit to be claimed.
+		/// - signature: The signature of the account that is claiming the spirit. //Sr25519Signature
+		/// - metadata: The metadata of the account that is claiming the spirit.
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
+		#[transactional]
+		pub fn claim_spirit(
+			origin: OriginFor<T>,
+			serial_id: SerialId,
+			signature: sr25519::Signature,
+			metadata: BoundedVec<u8, T::StringLimit>,
+		) -> DispatchResult {
+			ensure!(CanClaimSpirits::<T>::get(), Error::<T>::ClaimIsOver);
+			let sender = ensure_signed(origin)?;
+			let overlord = <Overlord<T>>::get();
+			match overlord {
+				None => Err(Error::<T>::OverlordNotSet.into()),
+				Some(overlord) => {
+					// Has the SerialId already been claimed
+					ensure!(
+						!ClaimedSpirits::<T>::contains_key(serial_id),
+						Error::<T>::SpiritAlreadyClaimed
+					);
+					// Check if valid SerialId to claim a spirit
+					ensure!(
+						Self::verify_claim(
+							sender.clone(),
+							metadata.clone(),
+							signature
+						),
+						Error::<T>::ClaimVerificationFailed
+					);
+					// Mint new Spirit and transfer to sender
+					pallet_rmrk_core::Pallet::<T>::mint_nft(
+						Origin::<T>::Signed(overlord).into(),
+						sender.clone(),
+						SPIRIT_COLLECTION_ID,
+						None,
+						None,
+						metadata,
+					)?;
+					ClaimedSpirits::<T>::insert(serial_id, true);
+
+					Self::deposit_event(Event::SpiritClaimed {
+						serial_id,
+						owner: sender,
+					});
+
+					Ok(())
+				}
+			}
+		}
+
+		/// Buy a rare egg of either type Legendary or Founder. Both Egg types will have a set
+		/// price. These will also be limited in quantity and on a first come, first serve basis.
+		///
+		/// Parameters:
+		/// - origin: The origin of the extrinsic.
+		/// - egg_type: The type of egg to be purchased.
+		/// - race: The race of the egg chosen by the user.
+		/// - career: The career of the egg chosen by the user or auto-generated based on metadata
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
+		#[transactional]
+		pub fn buy_rare_egg(
+			origin: OriginFor<T>,
+			egg_type: EggType,
+			race: RaceType,
+			career: CareerType,
+			metadata: BoundedVec<u8, T::StringLimit>,
+		) -> DispatchResult {
+			ensure!(CanPurchaseRareEggs::<T>::get(), Error::<T>::RareEggPurchaseNotAvailable);
+			let sender = ensure_signed(origin.clone())?;
+			let overlord = <Overlord<T>>::get();
+			match overlord {
+				None => Err(Error::<T>::OverlordNotSet.into()),
+				Some(overlord) => {
+					// Get Egg Price based on EggType
+					let egg_price = match egg_type {
+						EggType::Founder => {
+							T::FounderEggPrice::get()
+						},
+						EggType::Legendary => {
+							T::LegendaryEggPrice::get()
+						},
+						_ => T::NormalEggPrice::get(),
```

Comment:
We can return an error now. Then there's no need to compare `egg_price` and `NormalEggPrice` in the next line